### PR TITLE
add cherry-pick-pull from upstream along with docs for backporting

### PR DESCRIPTION
**Pull request by jaxesn**
*Issue #, if available:*

*Description of changes:*
We pulled this from upstream for use in eks-anywhere-build-tooling.  You can see an example PR that was opened using the script here https://github.com/aws/eks-anywhere-build-tooling/pull/61.

Added doc to the docs/developer folder, not sure if this is the best spot for development docs or not?  This obviously doesn't need to show up on the docs website.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/643#" title="Author self-approved">jaxesn</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [jaxesn]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: docs/developer/cherry-picks.md
Comment (taneyland) [inline docs/developer/cherry-picks.md:21]:
```suggestion
	points to your fork at `<github_user>/eks-anywhere`
```

**Review by abhay-krishna — APPROVED** [review]
/lgtm

**Pull request merged**
